src/main/java/mate/academy/springboot/swagger/controller/ProductController.java

```diff
@@ -0,0 +1,170 @@
+        if (productRequestDto == null) {
+            return new ResponseEntity<>(null, HttpStatus.BAD_REQUEST);
+        }
+        Product product;
+        try {
+            product = productService.save(productMapper.mapToModel(productRequestDto));
+        } catch (NoSuchElementException e) {
+            return new ResponseEntity<>(null, HttpStatus.CONFLICT);
+        }
+        return new ResponseEntity<>(product, HttpStatus.CREATED);
+    }
+
+    @Operation(summary = "Get a Product by its id")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Found the Product",
+                    content = { @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = ProductResponseDto.class)) }),
+            @ApiResponse(responseCode = "400", description = "Invalid id supplied",
+                    content = @Content),
+            @ApiResponse(responseCode = "404", description = "Product not found",
+                    content = @Content) })
+    @GetMapping("/{id}")
+    public ProductResponseDto get(@PathVariable("id")
+                                      @Parameter(name = "id", description = "Product id",
+                                              example = "1") Long id) {
+        return productMapper.mapToDto(productService.getById(id));
+    }
+
+    @Operation(summary = "Delete Product by its id")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "204", description = "Delete the Product",
+                    content = { @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = Product.class)) }),
+            @ApiResponse(responseCode = "400", description = "Invalid id supplied",
+                    content = @Content),
+            @ApiResponse(responseCode = "404", description = "Product not found",
+                    content = @Content) })
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable("id")
+                           @Parameter(name = "id",
+                                   description = "Product id", example = "1") Long id) {
+        productService.delete(id);
+    }
+
+    @Operation(summary = "Update an existing product",
+            description = "update info about product", tags = { "product" })
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "successful operation"),
+            @ApiResponse(responseCode = "400", description = "Invalid ID supplied"),
+            @ApiResponse(responseCode = "404", description = "Product not found"),
+            @ApiResponse(responseCode = "405", description = "Validation exception") })
+    @PutMapping(value = "/products/{id}", consumes = { "application/json" })
+    public void update(
+            @Parameter(description = "Id of the product to be update. Cannot be empty.",
```

Comment:
It can not be empty by the definition, it is part of URL

Reply:
ok